src/app/clusters/resource-monitoring-server/resource-monitoring-server.cpp

```diff
@@ -232,6 +242,32 @@ CHIP_ERROR Instance::EnumerateAcceptedCommands(const ConcreteClusterPath & clust
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR Instance::ReadReplacableProductList(AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err;
+    if (Instance::HasFeature(ResourceMonitoring::Feature::kReplacementProductList))
+    {
+        ReplacementProductListManager * productListManagerInstance = Instance::GetReplacementProductListManagerInstance();
+        if (nullptr == productListManagerInstance)
+        {
+            aEncoder.EncodeEmptyList();
+            return CHIP_NO_ERROR;
+        }
+
+        productListManagerInstance->Reset();
+
+        err = aEncoder.EncodeList([&](const auto & encoder) -> CHIP_ERROR {
+            Attributes::ReplacementProductStruct::Type replacementProductStruct;
+            while (productListManagerInstance->Next(replacementProductStruct) == CHIP_NO_ERROR)
```

Comment:
Why is this dropping errors on the floor?  If the callee returns some "I am in a bad state" error, should we not return that instead of silently truncating the list?